GEODE-2155 Auto-reconnect fails with NPE

I created a test to reproduce the problem by introducing a non-Declarable
cache listener and then initiating a forced-disconnect.  cache.xml is
generated and when it's used to rebuild the cache a CacheXmlException
is thrown, reliably reproducing the problem.

The fix is to simply catch the CacheXmlException in
InternalDistributedSystem.reconnect() and terminate reconnection attempt

+import org.apache.geode.cache.CacheXmlException;
+            } catch (CacheXmlException e) {
+              logger.warn("Exception occured while trying to create the cache during reconnect", e);
+              reconnectDS.disconnect();
+              reconnectDS = null;
+              reconnectCancelled = true;
+              break;

INS26 INS40 INS12 INS44 INS8 INS43 INS42 INS21 INS21 INS21 INS21 INS10 INS42 INS32 INS32 INS7 INS7 INS42 INS42 INS45 INS42 INS42 INS42 INS42 INS33 INS42 INS9